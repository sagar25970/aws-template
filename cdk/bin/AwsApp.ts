import { App, AppProps } from 'aws-cdk-lib';

export class AwsApp extends App {

    constructor(props?: AppProps) {
        super(props);
    }
}