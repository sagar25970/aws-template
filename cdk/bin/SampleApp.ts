import { App, AppProps } from 'aws-cdk-lib';

export class SampleApp extends App {

    constructor(props?: AppProps) {
        super(props);
    }
}