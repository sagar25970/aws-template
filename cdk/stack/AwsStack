import { App, Stack, StackProps } from "aws-cdk-lib";

export interface AwsStackProps extends StackProps {
    stackProp: string;
}

export class AwsStack extends Stack {
    constructor(scope: App, id: string, props: AwsStackProps) {
        super(scope, id, props);
    }
}