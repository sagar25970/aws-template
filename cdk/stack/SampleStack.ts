import { App, Stack, StackProps } from "aws-cdk-lib";
import { Key } from "aws-cdk-lib/aws-kms";
import { Bucket, BucketEncryption } from "aws-cdk-lib/aws-s3";
import { Construct } from "constructs";
import { createSampleEncryptedBucket } from "../construct/bucket/SampleBucket";
import { NodejsFunction } from "aws-cdk-lib/aws-lambda-nodejs";

export interface SampleStackProps extends StackProps {
    nameSpace: string;
}

export class SampleStack extends Stack {
    constructor(scope: App, id: string, props: SampleStackProps) {
        super(scope, id, props);

        const { bucket, bucketEncryptionKey } = createSampleEncryptedBucket(this, props);
    }
}