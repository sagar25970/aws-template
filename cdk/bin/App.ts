import { AwsApp } from "./AwsApp";

const awsApp = new AwsApp();