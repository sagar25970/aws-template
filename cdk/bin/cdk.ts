import { SampleApp } from "./SampleApp";

const sampleApp = new SampleApp();
sampleApp.synth();