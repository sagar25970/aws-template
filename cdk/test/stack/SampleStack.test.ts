import { describe, it, expect } from '@jest/globals';
import { Template } from 'aws-cdk-lib/assertions';
import { SampleStack } from '../../stack/SampleStack';
import { SampleApp } from '../../bin/SampleApp';
import { App, Stack } from 'aws-cdk-lib';
import { SynthUtils } from '@aws-cdk/assert';
import * as fs from 'fs';

const sampleStackProps = {
    env: {
        account: '078900383518',
        region: 'ap-south-1'
    },
    stackName: `demo-sample-stack`,
    nameSpace: 'demo'
}

describe('Sample Stack', () => {

    let app: App;
    let sampleStack: Stack;
    let template: Template;

    beforeEach(() => {
        app = new SampleApp();
        sampleStack = new SampleStack(app, 'demo-sample-stack', sampleStackProps);
        template = Template.fromStack(sampleStack);
        fs.writeFile('template.json', JSON.stringify(template.toJSON()), () => { });
    });

    it('should match all resources', () => {
        expect(SynthUtils.toCloudFormation(sampleStack)).toMatchSnapshot();
    });

    it('should create S3 Bucket', () => {
        template.resourceCountIs('AWS::S3::Bucket', 1);
        template.hasResourceProperties('AWS::S3::Bucket', {
            BucketName: 'demo-sample-bucket'
        });
    });

    it('should create kms encryption key', () => {
        template.resourceCountIs('AWS::KMS::Key', 1);
        template.hasResourceProperties('AWS::KMS::Key', {
            Description: 'demo sample encryption key'
        });
    });
});