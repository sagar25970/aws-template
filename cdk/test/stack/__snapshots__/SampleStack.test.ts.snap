// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sample Stack should match all resources 1`] = `
{
  "Resources": {
    "demosamplebucket1812FC82": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "BucketEncryption": {
          "ServerSideEncryptionConfiguration": [
            {
              "ServerSideEncryptionByDefault": {
                "KMSMasterKeyID": {
                  "Fn::GetAtt": [
                    "demosampleencryptionkeyAEDAC18A",
                    "Arn",
                  ],
                },
                "SSEAlgorithm": "aws:kms",
              },
            },
          ],
        },
        "BucketName": "demo-sample-bucket",
        "VersioningConfiguration": {
          "Status": "Enabled",
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
    "demosampleencryptionkeyAEDAC18A": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "Description": "demo sample encryption key",
        "KeyPolicy": {
          "Statement": [
            {
              "Action": "kms:*",
              "Effect": "Allow",
              "Principal": {
                "AWS": {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::078900383518:root",
                    ],
                  ],
                },
              },
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::KMS::Key",
      "UpdateReplacePolicy": "Retain",
    },
  },
}
`;
